### Задача №1

#### Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти ширину доверительного интервала для оценки математического ожидания `mu` с надёжностью 0.95, если выборочное среднее равно 80, а объём выборки равен 256.

In [1]:
import numpy as np
from scipy import stats
import scipy

По условиям задачи рассматриваемая величина имеет нормальное равномерное распределение. $$Z = \dfrac{\overline{X} - \mu}{\sigma / \sqrt{n}}$$ Обозначим $\alpha = 1 - p$. Можно убедиться в том, что
$$P \left( t_{\alpha / 2} \leq Z \leq t_{1 - \alpha / 2} \right) = p,$$
где $t_{x}$ — квантиль порядка $x$ для стандартного нормального распределения. 

Доверительный интервал для параметра $\mu$ будет рассчитываться по формуле:  $$P \left( \overline{X} + t_{\alpha / 2} \cdot \dfrac{\sigma}{\sqrt{n}} \leq \mu \leq \overline{X} + t_{1 - \alpha / 2} \cdot \dfrac{\sigma}{\sqrt{n}} \right) = p$$

In [2]:
mean = 80
n = 256
p = 0.95
alpha = 1 - p
scale = 16
t1 = stats.norm.ppf(alpha / 2)
t2 = stats.norm.ppf(1 - alpha / 2)
t1, t2

(-1.959963984540054, 1.959963984540054)

Доверительный интервал:

In [3]:
a = (mean + t1 * scale / np.sqrt(n))
b = (mean + t2 * scale / np.sqrt(n))
print(f'Ширина доверительного интервала: {round((b-a), 8)}')

Ширина доверительного интервала: 3.91992797


### Задача №2

#### Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
```
202, 203, 199, 197, 195, 201, 200, 204, 194, 190
```
#### Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%? Найдите P-значение для данного теста и укажите его в качестве ответа.

Из условия задачи следует, что проверяется гипотеза относительно математического ожидания нормально распределённой случайной величины с не известной дисперсией. В этом случае будет использоваться __t-статистика__:
$$T = \dfrac{\overline{X} - \mu}{\sigma_X / \sqrt{n}},$$
где $\sigma_X$ — несмещённая оценка среднего квадратического отклонения.

В предположении верности гипотезы $H_0$ t-статистика имеет __распределение Стьюдента__ или __t-распределение__ с параметром $df = n - 1$.

In [4]:
weight = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
mean = weight.mean()
std = weight.std(ddof=1)
mean, std

(198.5, 4.453463071962462)

В нашем случае статистика равна: $$T = \dfrac{mean - 200}{std / \sqrt{10}},$$

In [5]:
n = 10
T=(mean - 200)/(std/np.sqrt(n))
T

-1.0651074037450896

Находим критическую область:

In [6]:
alpha = 0.01

t1 = stats.t.ppf(alpha / 2, df=n - 1)
t2 = stats.t.ppf(1 - alpha / 2, df=n - 1)

t1, t2

(-3.2498355440153697, 3.2498355440153697)

В итоге критическая область равна:
$$\Omega_\alpha = (-\infty, -3.25) \cup (3.25,  \infty)$$ Таким образом видим, что значение Т не попадает в критическую область, значит продавец прав

##### Находим Р значение

т.к. у нас двухсторонняя область, Р-значение будем находить по формуле: $$P = 2 \cdot \min (P_l, P_r)$$

In [7]:
p_left = stats.t.cdf(T, df=n - 1)
p_right = 1 - stats.t.cdf(T, df=n - 1)

pvalue = 2 * min(p_left, p_right)

print(f'Р-значение: {round(pvalue, 8)}')

Р-значение: 0.3145666


### Задача №3

#### Дана выборка диаметров подшипников из примера 1 с занятия 5:
```
samples = [0.6603, 0.9466, 0.5968, 1.3792, 1.5481, 0.7515, 1.0681, 1.1134,
           1.2088, 1.701 , 1.0282, 1.3579, 1.0191, 1.1784, 1.1168, 1.1372,
           0.7273, 1.3958, 0.8665, 1.5112, 1.161 , 1.0232, 1.0865, 1.02  ]
```
Предполагая, что диаметры подшипников распределены нормально, проверьте гипотезу о том, что дисперсия случайной величины равна `0.0625` при уровне значимости `alpha = 0.05`. Что для этого нужно знать:
1. Альтернативная гипотеза двухсторонняя.
2. Статистика для теста: `H = (n - 1) * sample_variance / variance`, где `n` - число элементов в выборке, `sample_variance` - несмещённая оценка дисперсии, `variance` - утверждаемая нулевой гипотезой дисперсия.
3. Эта статистика в предположении верности нулевой гипотезы имеет распределение хи-квадрат с параметром `df = n - 1`. Её квантили можно найти с помощью функции `scipy.stats.chi2.ppf`.

В качестве ответа запишите 1, если нулевая гипотеза верна, и 0 иначе.

In [8]:
samples = np.array([0.6603, 0.9466, 0.5968, 1.3792, 1.5481, 0.7515, 1.0681, 1.1134,
           1.2088, 1.701 , 1.0282, 1.3579, 1.0191, 1.1784, 1.1168, 1.1372,
           0.7273, 1.3958, 0.8665, 1.5112, 1.161 , 1.0232, 1.0865, 1.02  ])

In [9]:
mean = samples.mean()
std = samples.std(ddof=1)
n = samples.shape[0]
mean, std, n

(1.1084541666666665, 0.27936526343958135, 24)

Статистика для теста, согласно условиям задачи, определяется по формуле: $$H = \dfrac{(n - 1) \cdot sample variance }{variance}$$ где n - число элементов в выборке, sample_variance - несмещённая оценка дисперсии, variance - утверждаемая нулевой гипотезой дисперсия

In [10]:
sample_variance = std**2
variance = 0.0625
H = ((n - 1) * sample_variance) / variance
H

28.720541753333343

Находим критическую область

In [11]:
alpha = 0.05

t1 = scipy.stats.chi2.ppf(alpha / 2, df=n - 1)
t2 = scipy.stats.chi2.ppf(1 - alpha / 2, df=n - 1)

t1, t2

(11.688551922452438, 38.0756272503558)

В итоге критическая область равна:
$$\Omega_\alpha = (-\infty, 11.69) \cup (38.07,  \infty)$$ Таким образом видим, что значение H не попадает в критическую область, значит нулевая гипотеза верна

#### Ответ: 1